File: javascript/image_moderation_batch_jobs_aksk_demo.js
/**
 * 图像内容检测批量异步服务ak,sk请求方式的使用示例
 */
var content = require("./ais_sdk/image_moderation_batch_jobs");
var utils = require("./ais_sdk/utils");

// 初始化服务的区域信息，目前支持华北-北京一(cn-north-1)、亚太-香港(ap-southeast-1)等区域信息
utils.initRegion("cn-north-1");

var app_key = "*************";
var app_secret = "*************";

var filepath = "./data/moderation-terrorism.jpg";
var data = utils.changeFileToBase64(filepath);

// obs链接需要和region区域一致，不同的region的obs资源不共享
url1 = "https://ais-sample-data.obs.cn-north-1.myhuaweicloud.com/terrorism.jpg";
url2 = "https://ais-sample-data.obs.cn-north-1.myhuaweicloud.com/antiporn.jpg";

content.batch_jobs_aksk(app_key, app_secret, [url1, url2], ["politics", "terrorism", "porn"], function (result) {
    console.log(result);
});

File: python/image_moderation_batch_jobs_aksk_demo.py
# -*- coding:utf-8 -*-
from ais_sdk.image_moderation_batch_jobs import image_batch_jobs_aksk
from ais_sdk.utils import init_global_env

if __name__ == '__main__':
    # Services currently support North China-Beijing 1 (cn-north-1), Asia Pacific-Hong Kong (ap-southeast-1)
    init_global_env('cn-north-1')

    #
    # access moderation image of batch jobs,post data by ak,sk
    #
    app_key = '*************'
    app_secret = '************'

    demo_data_url1 = 'https://ais-sample-data.obs.cn-north-1.myhuaweicloud.com/terrorism.jpg'
    demo_data_url2 = 'https://ais-sample-data.obs.cn-north-1.myhuaweicloud.com/antiporn.jpg'

    # call interface use the url
    result = image_batch_jobs_aksk(app_key, app_secret, [demo_data_url1, demo_data_url2], ['politics','terrorism'])
    print result


